yaksh/models.py

```diff
@@ -335,20 +334,32 @@ def create_demo_quiz(self, user):
         return demo_quiz
 
     def get_total_students(self, course):
+        try:
+            qp = self.questionpaper_set.get().id
+        except QuestionPaper.DoesNotExist:
+            qp = None
         return AnswerPaper.objects.filter(
-                question_paper=self.questionpaper_set.get().id,
+                question_paper=qp,
                 course=course
             ).values_list("user", flat=True).distinct().count()
 
     def get_passed_students(self, course):
+        try:
+            qp = self.questionpaper_set.get().id
+        except QuestionPaper.DoesNotExist:
+            qp = None
         return AnswerPaper.objects.filter(
-                question_paper=self.questionpaper_set.get().id,
+                question_paper=qp,
                 course=course, passed=True
             ).values_list("user", flat=True).distinct().count()
 
     def get_failed_students(self, course):
+        try:
+            qp = self.questionpaper_set.get().id
+        except QuestionPaper.DoesNotExist:
+            qp = None
```

Comment:
It would be good to add tests for try catch in `get_failed_students`, `get_passed_students`, `get_total_students`.